runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java

```diff
@@ -261,6 +295,44 @@ public void onItemQuantityChanged(ItemQuantityChanged itemQuantityChanged)
 		}
 	}
 
+	@Subscribe
+	public void onNpcLootReceived(final NpcLootReceived npcLootReceived)
+	{
+		final NPC npc = npcLootReceived.getNpc();
+		final Collection<ItemStack> items = npcLootReceived.getItems();
+
+		for (ItemStack i: items)
+		{
+			lootItems.put(i.getId(), npc.getLocalLocation());
+		}
+	}
+
+	@Subscribe
+	public void onPlayerLootReceived(final PlayerLootReceived playerLootReceived)
+	{
+		final Player player = playerLootReceived.getPlayer();
+		final Collection<ItemStack> items = playerLootReceived.getItems();
+
+		for (ItemStack i: items)
+		{
+			lootItems.put(i.getId(), player.getLocalLocation());
+		}
+	}
+
+	boolean isPlayerOwned(GroundItem groundItem)
```

Comment:
This is too expensive to be called per item per frame. You should probably figure out how to mark `GroundItem`s instead.